sheet.go

```diff
@@ -78,10 +78,22 @@ func (s *Sheet) Col(idx int) *Col {
 // containing the data from the field "A1" on the spreadsheet.
 func (sh *Sheet) Cell(row, col int) *Cell {
 
-	if len(sh.Rows) > row && sh.Rows[row] != nil && len(sh.Rows[row].Cells) > col {
-		return sh.Rows[row].Cells[col]
+	// If the user requests a row beyond what we have, then extend.
+	if len(sh.Rows) <= row {
+		numneeded := (len(sh.Rows) - row) + 1
+		for x := 0; x < numneeded; x++ {
+			sh.AddRow()
+		}
+	}
+	r := sh.Rows[row]
+
+	if len(r.Cells) < (col + 1) {
+		numneeded := (col - len(r.Cells)) + 1
+		for x := 0; x < numneeded; x++ {
+			r.AddCell()
+		}
 	}
-	return new(Cell)
+	return r.Cells[col]
```

Comment:
I actually use something similar, but I believe this could read a bit easier like so:

```golang
for len(sh.Rows) <= row {
	sh.AddRow()
}
r := sh.Rows[row]
for len(r.Cells) <= col {
	r.AddCell()
}
```